SDRNA RNA Mining.ipynb

```diff
@@ -0,0 +1,278 @@
+    "and loops through each Genbank files to look for the rRNA feature.\n",
+    "\"\"\"\n",
+    "\n",
+    "def SDRNA_RNA_Mining(folder,subfolder):\n",
+    "    pathname = Path(folder + \"/\" + subfolder + \"/\")\n",
+    "    filelist = os.listdir(pathname)\n",
+    "    opname = folder + \" \" + subfolder + \" output.fasta\"\n",
+    "    output_file = pathname / opname\n",
+    "    f = open(output_file, \"w\")\n",
+    "    \n",
+    "    for fn in range(len(filelist)):\n",
+    "        if \"gb\" in filelist[fn]: \n",
+    "            gbfile = filelist[fn]\n",
+    "            input_file = pathname / gbfile\n",
+    "\n",
+    "            for rec in SeqIO.parse(input_file, \"gb\"): \n",
+    "                acc = rec.annotations['accessions'][0]\n",
+    "                seq = rec.seq\n",
+    "\n",
+    "                for feature in rec.features:\n",
+    "                    ifrrRNA = 0\n",
+    "\n",
+    "                    if \"rRNA\" in feature.type:\n",
+    "                        print(feature)\n",
+    "\n",
+    "                        seqstart = feature.location.start.position\n",
+    "                        seqend = feature.location.end.position\n",
+    "\n",
+    "                        if feature.location.strand == 1:\n",
+    "                            strdir = \"forward\"\n",
+    "                            rRNAseq = seq[seqstart:seqend]\n",
+    "                        elif feature.location.strand == -1:\n",
+    "                            strdir = \"revcomp\"\n",
+    "                            rRNAseq = seq[seqstart:seqend].reverse_complement()[::]\n",
+    "                        else:\n",
+    "                            strdir = \"\"\n",
+    "\n",
+    "                        if \"db_xref\" in feature.qualifiers:\n",
+    "                            addinfo1 = feature.qualifiers[\"db_xref\"][0]\n",
+    "                        else:\n",
+    "                            addinfo1 = \"\"\n",
+    "                        \n",
+    "                        if \"product\" in feature.qualifiers:\n",
+    "                            addinfo2 = feature.qualifiers[\"product\"][0]\n",
+    "                        elif \"note\" in feature.qualifiers:\n",
+    "                            addinfo2 = feature.qualifiers[\"note\"][0]\n",
+    "                        else:\n",
+    "                            addinfo2 = \"\"\n",
+    "\n",
+    "                        f.write(\">{0}|{1}|{2}-{3}|{4}|{5}|{6}| \\n{7}\\n\".format(acc, \n",
+    "                                                                               feature.qualifiers[\"locus_tag\"][0], \n",
+    "                                                                               seqstart, seqend, strdir,\n",
+    "                                                                               addinfo1, addinfo2, rRNAseq))\n",
+    "    f.close()\n",
+    "    \n",
+    "SDRNA_RNA_Mining(\"SC 7000071044\",\"NCBI SC S288C\")"
```

Comment:
I'd also suggest to write a few unit tests for this function - to verify expected behaviors with biological relevance. As an example - yeast biology would suggest to find:
- the small, 40S subunit - which contains one rRNA: 18S, 1800 nt
- the large, 60S subunit - that includes three rRNAs: 5S, 121 nt; 5.8S, 158 nt; and 25S, 3396 nt
Those are good candidates for code verification.

Reply:
I actually have the length in the oligo designing notebook and I was thinking of adding a finding consensus sequence function in between and maybe it will be a good thing to add it there?

Reply:
Added rRNA length in [rRNA_mining.ipynb](https://github.com/Zymergen/adv-tech-prog/blob/FZL_merge_on_ATP/Code/rRNA_mining.ipynb) on atp-tech-prog --> FZL_merge_on_ATP branch --> Code